kernel/kernel-api/src/main/java/io/delta/kernel/internal/rowtracking/MaterializedRowTrackingColumn.java

```diff
@@ -145,6 +148,39 @@ public static Optional<Metadata> assignMaterializedColumnNamesIfNeeded(Metadata
         : Optional.of(metadata.withMergedConfiguration(configsToAdd));
   }
 
+  /**
+   * Converts a logical column to its corresponding physical column if it is a row tracking metadata
+   * column (ROW_ID or ROW_COMMIT_VERSION).
+   *
+   * <p>Note that we must not mark the physical columns as metadata columns because as far as the
+   * parquet reader is concerned, these columns are not metadata columns.
+   *
+   * @param column The logical column to convert.
+   * @param metadata The current metadata of the table.
+   * @return An Optional containing the physical column if conversion is applicable;
+   *     Optional.empty() otherwise.
+   */
+  public static Optional<StructField> convertToPhysicalColumn(
+      StructField column, Metadata metadata) {
+    if (column.equals(METADATA_ROW_ID_COLUMN)) {
```

Comment:
We'll need to fail if a Row Tracking metadata column is requested but row tracking isn't enabled on the table. You'll need to check that the rowTracking table feature is supported in the table `Protocol`. Also add a test for this case